cocotb/handle.py

```diff
@@ -656,13 +656,16 @@ def _set_value(self, value, call_sim):
         """
         value, set_action = self._check_for_set_action(value)
 
-        if isinstance(value, int) and value < 0x7fffffff and len(self) <= 32:
-            call_sim(self._handle.set_signal_val_long, set_action, value)
+        if isinstance(value, int):
+            if value >= -0x80000000 and value <= 0x7fffffff and len(self) <= 32:
+                call_sim(self._handle.set_signal_val_long, set_action, value)
+            else:
+                value = value.to_bytes((len(self)+7)//8, byteorder='big', signed=value<0)
```

Comment:
This line is problematic. Defaulting to big endian is probably not a good idea, most systems are little endian nowadays. We could use `sys.byteorder`, but that makes an assumption that the user knows the byte order of their system. We might end up in a situations where an unexpected value is written.

Reply:
Actually, I think we want to hardcode this to big endian, as that is how we want to present it to the simulator. From the docs:
```The byteorder argument determines the byte order used to represent the integer. If byteorder is "big", the most significant byte is at the beginning of the byte array. If byteorder is "little", the most significant byte is at the end of the byte array. To request the native byte order of the host system, use sys.byteorder as the byte order value.```

Reply:
Endianness isn't a matter of "correct", it's a matter of preference, and the reality of the hardware being used.

Reply:
I think there is some confusion here with big/little endianness vs ascending/descending vector range. 
```set_signal_value_long()``` is used for setting vectors < 32-bit and assumes that the vector we're setting has a "descending" range. I wanted to ensure consistency with this behaviour for setting vectors > 32-bit using ```set_signal_value_bytes()``` 
Here's a code snippet that illustrates that we want to use big (network) endian to ensure the bit pattern remains as intended left to right and avoid any extra work with reordering of the bytes in the C++ code.
```python
In [1]: N=0x0123456789abcdef

In [2]: N.to_bytes(10, byteorder="big", signed=False).hex()
Out[2]: '00000123456789abcdef'

In [3]: N.to_bytes(10, byteorder="little", signed=False).hex()
Out[3]: 'efcdab89674523010000'

In [4]: N.to_bytes(10, byteorder="big", signed=False).hex()[::-1]
Out[4]: 'fedcba98765432100000'

In [5]: N.to_bytes(10, byteorder="little", signed=False).hex()[::-1]
Out[5]: '00001032547698badcfe'
```